Pod/Classes/VOKCoreDataManager.m

```diff
@@ -171,47 +154,72 @@ - (void)initPersistentStoreCoordinator
         storeType = NSSQLiteStoreType;
     }
     
-    if (![_persistentStoreCoordinator addPersistentStoreWithType:storeType
-                                                   configuration:nil
-                                                             URL:storeURL
-                                                         options:options
-                                                           error:&error]) {
-        if (self.migrationFailureOptions == VOKMigrationFailureOptionWipeRecoveryAndAlert) {
-            [[[UIAlertView alloc] initWithTitle:@"Migration Failed"
-                                        message:@"Migration has failed, data will be erased to ensure application stability."
-                                       delegate:nil
-                              cancelButtonTitle:@""
-                              otherButtonTitles:nil] show];
-        }
-        
-        if (self.migrationFailureOptions == VOKMigrationFailureOptionWipeRecoveryAndAlert ||
-            self.migrationFailureOptions == VOKMigrationFailureOptionWipeRecovery) {
-            VOK_CDLog(@"Full database delete and rebuild");
-            [[NSFileManager defaultManager] removeItemAtPath:storeURL.path error:nil];
-            if (![_persistentStoreCoordinator addPersistentStoreWithType:storeType
-                                                           configuration:nil
-                                                                     URL:storeURL
-                                                                 options:nil
-                                                                   error:&error]) {
-                [NSException raise:@"Vokoder Persistant Store Creation Failure after migration"
-                            format:@"Unresolved error %@, %@", error, [error userInfo]];
+    __block NSError *error;
+    BOOL (^createPersistantStore)() = ^BOOL() {
+        NSError *localError;
+        BOOL success =  [_persistentStoreCoordinator addPersistentStoreWithType:storeType
+                                                                  configuration:nil
+                                                                            URL:storeURL
+                                                                        options:options
+                                                                          error:&localError];
+        error = localError;
+        return success;
+    };
+    
+    if (!createPersistantStore()) {
```

Comment:
I know I might sound like a dummy, but I think the use of the block here is necessarily complex. Why not simply attempt to create the persistent store, check the bool, and display the rror in the old fashioned way?